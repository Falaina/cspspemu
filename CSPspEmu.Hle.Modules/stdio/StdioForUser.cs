using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSPspEmu.Hle.Modules.stdio
{
	public class StdioForUser : HleModuleHost
	{
	}
}
