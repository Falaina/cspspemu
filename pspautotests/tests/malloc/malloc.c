#include <pspkernel.h>
#include <pspthreadman.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include <common.h>


#define NUM_BUFFERS 64
#define BUFFER_SIZE (5 * 1024 * 1024 / NUM_BUFFERS)

// 24MB to use
int main(int argc, char** argv) {
	unsigned int sum;
	int n, m;
	unsigned int *buffers[NUM_BUFFERS];
	unsigned int *buffer;
	char *temp;
	printf("STARTED\n");
	{
		for (n = 0; n < NUM_BUFFERS; n++) {
			buffers[n] = calloc(BUFFER_SIZE, sizeof(unsigned int));
			buffer = buffers[n];
			printf("%s\n", (buffer == NULL) ? "NULL" : "NOT NULL");
			for (m = 0; m < BUFFER_SIZE; m++) buffer[m] = n;
		}
		for (n = 0; n < NUM_BUFFERS; n++) {
			sum = 0;
			buffer = buffers[n];
			for (m = 0; m < BUFFER_SIZE; m++) sum += buffer[m];
			printf("%d: %d\n", n, sum);
		}
		temp = malloc(24 * 1024 * 1024);
		printf("%s\n", (temp == NULL) ? "NULL" : "NOT NULL");
		if (temp != NULL) free(temp);
		for (n = 0; n < NUM_BUFFERS; n++) {
			free(buffers[n]);
		}
		temp = malloc(24 * 1024 * 1024);
		printf("%s\n", (temp == NULL) ? "NULL" : "NOT NULL");
		if (temp != NULL) free(temp);
	}
	printf("ENDED\n");

	return 0;
}
