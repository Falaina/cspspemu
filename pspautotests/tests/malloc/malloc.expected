STARTED
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
NOT NULL
0: 0
1: 81920
2: 163840
3: 245760
4: 327680
5: 409600
6: 491520
7: 573440
8: 655360
9: 737280
10: 819200
11: 901120
12: 983040
13: 1064960
14: 1146880
15: 1228800
16: 1310720
17: 1392640
18: 1474560
19: 1556480
20: 1638400
21: 1720320
22: 1802240
23: 1884160
24: 1966080
25: 2048000
26: 2129920
27: 2211840
28: 2293760
29: 2375680
30: 2457600
31: 2539520
32: 2621440
33: 2703360
34: 2785280
35: 2867200
36: 2949120
37: 3031040
38: 3112960
39: 3194880
40: 3276800
41: 3358720
42: 3440640
43: 3522560
44: 3604480
45: 3686400
46: 3768320
47: 3850240
48: 3932160
49: 4014080
50: 4096000
51: 4177920
52: 4259840
53: 4341760
54: 4423680
55: 4505600
56: 4587520
57: 4669440
58: 4751360
59: 4833280
60: 4915200
61: 4997120
62: 5079040
63: 5160960
NULL
NOT NULL
ENDED